src/transfers-history/services/SpokePoolEventsQueryService.ts

```diff
@@ -78,7 +78,16 @@ export class SpokePoolEventsQueryService {
       return { ...acc, [event.blockNumber]: true };
     }, {} as Record<number, any>);
     const uniqueBlockNumbersList = Object.keys(uniqueBlockNumbers).map((blockNumber) => parseInt(blockNumber));
-    const blocks = await Promise.all(uniqueBlockNumbersList.map((blockNumber) => this.provider.getBlock(blockNumber)));
+    this.logger.debug(
+      "[getBlocksTimestamp]",
+      `🟢 chain ${this.chainId}: fetching ${uniqueBlockNumbersList.length} blocks`
+    );
+    const blocksChunks = this.getArrayChunks(uniqueBlockNumbersList);
+    const blocks = [];
+    for (const blocksChunk of blocksChunks) {
+      const newBlocks = await Promise.all(blocksChunk.map((blockNumber) => this.provider.getBlock(blockNumber)));
```

Comment:
Why not use bluebird promises? AFAIK, you can specify the max number of promises you want in flight. I think bluebird might be more efficient (no need to wait on _all_ the requests in the prev group to finish). It also might be a bit clearer since it looks just like a typical Promise.all.

Not strongly opinionated though. If there's some reason the current strategy is preferable, that's totally fine too.

Reply:
I can use bluebird ofc, but this time I actually preferred to wait on all the requests in the prev group to finish because I got fluctuating performances from Infura on some chains (especially on optimism). Some times a huge number of requests went through, but other times the performance was quite bad, so I preferred to not force it by having the max number of promises in flight all the time